src/main/java/org/alfresco/repo/security/authority/AuthorityDAOImpl.java

```diff
@@ -1046,6 +1046,17 @@ else if (type.isPrefixed())
             final String authorityName = prefix + shortName;
             if (shortName.startsWith(prefix))
             {
+                if (type == AuthorityType.GROUP)

```

Comment:
Are these characters allowed for users?

Reply:
I thought of initially adding it for the `AuthorityType.USER` but the [createAuthority](https://github.com/Alfresco/alfresco-repository/blob/master/src/main/java/org/alfresco/repo/security/authority/AuthorityServiceImpl.java#L645) method calls another method [checkTypeIsMutable](https://github.com/Alfresco/alfresco-repository/blob/master/src/main/java/org/alfresco/repo/security/authority/AuthorityServiceImpl.java#L527) and it has explicit check for either `AuthorityType.GROUP` or `AuthorityType.ROLE` which makes me to believe this `createAuthority` method may not be creating `AuthorityType.USER`.  Hence, I only guarded the `checkTypeIsMutable` method with `AuthorityType.Group`.

Reply:
I think it is required to extend the scope of this change to make sure that Users and Groups can not be created with wrong characters using Java APIs.